deflate_medium.c

```diff
@@ -85,13 +85,10 @@ static void insert_match(deflate_state *s, struct match match) {
     } else {
         match.strstart += match.match_length;
         match.match_length = 0;
-        if (match.strstart >= (MIN_MATCH - 2))
-#if MIN_MATCH != 3
-            functable.insert_string(s, match.strstart + 2 - MIN_MATCH, MIN_MATCH - 2);
-#else
-            functable.quick_insert_string(s, match.strstart + 2 - MIN_MATCH);
-#endif
-        /* If lookahead < MIN_MATCH, ins_h is garbage, but it does not
+        if (match.strstart >= (WANT_MIN_MATCH - 2))
+            functable.quick_insert_string(s, match.strstart + 2 - STD_MIN_MATCH);
```

Comment:
Not 100% confident this is correct...

Reply:
What do you think is wrong?

Reply:
I don't think it should use `WANT_MIN_MATCH`. It only needs 1 extra byte (`2 - 3 = -1`), but `WANT_MIN_MATCH - 2 = 2`.

Reply:
Yes, I think you are correct, running tests now to verify it does not break.